### PR TITLE
Java: Promote Sensitive Logging query

### PR DESCRIPTION
**Pull request by joefarebrother**

### CODE REVIEW

File: java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/** Provides configurations for sensitive logging queries. */
+
+import java
+import semmle.code.java.dataflow.ExternalFlow
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.security.SensitiveActions
+import DataFlow
+
+/** Variable keeps sensitive information judging by its name * */
```
Comment (smowton) [inline java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll:9]:
```suggestion
/** A variable that may hold sensitive information judging by its name. */
```

File: java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class CredentialExpr extends Expr {
+  CredentialExpr() {
+    exists(Variable v | this = v.getAnAccess() |
+      v.getName().regexpMatch([getCommonSensitiveInfoRegex(), "(?i).*(username).*"]) and
+      not v.isFinal()
+    )
+  }
+}
+
+class SensitiveLoggerConfiguration extends DataFlow::Configuration {
```
Comment (smowton) [inline java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll:19]:
```suggestion
/**
 * A data-flow configuration for identifying potentially-sensitive data flowing to a log output.
 */
class SensitiveLoggerConfiguration extends DataFlow::Configuration {
```

**Comment by smowton** [comment]
^^ Fixed my own broken suggestion

**Review by smowton — APPROVED** [review]

**Comment by atorralba** [comment]
We'll need to add the `@security-severity` tag to this query.

**Pull request merged**